src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts

```diff
@@ -421,11 +426,17 @@ export class UbsAdminOrderComponent implements OnInit, OnDestroy, AfterContentCh
 
     if (changedValues.orderDetailsForm) {
       changedValues.orderDetailDto = this.formatBagsValue(changedValues.orderDetailsForm);
+      const keyEcoNumberFromShop = 'ecoNumberFromShop';
       if (changedValues.orderDetailsForm.storeOrderNumbers) {
-        const keyEcoNumberFromShop = 'ecoNumberFromShop';
         changedValues[keyEcoNumberFromShop] = {
-          ecoNumber: changedValues.orderDetailsForm.storeOrderNumbers
+          ecoNumber: this.orderForm.value.orderDetailsForm.storeOrderNumbers
+        };
+        this.deleteNumberOrderFromEcoShop = false;
+      } else if (this.deleteNumberOrderFromEcoShop) {
+        changedValues[keyEcoNumberFromShop] = {
+          ecoNumber: this.orderForm.value.orderDetailsForm.storeOrderNumbers
```

Comment:
you can try to remove codesmell `if (changedValues.orderDetailsForm) {
  changedValues.orderDetailDto = this.formatBagsValue(changedValues.orderDetailsForm);
  const keyEcoNumberFromShop = 'ecoNumberFromShop';

  if (changedValues.orderDetailsForm.storeOrderNumbers || this.deleteNumberOrderFromEcoShop) {
    changedValues[keyEcoNumberFromShop] = {
      ecoNumber: this.orderForm.value.orderDetailsForm.storeOrderNumbers  
    };
    this.deleteNumberOrderFromEcoShop = false;
  }
}
`